apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -25,10 +27,22 @@
 
   @Nonnull ApolloCall<T> clone();
 
-  interface Callback<T> {
-    void onResponse(@Nonnull Response<T> response);
+  abstract class Callback<T> {
+    public abstract void onResponse(@Nonnull Response<T> response);
+
+    public abstract void onFailure(@Nonnull ApolloException e);
+
+    public void onHttpError(@Nonnull ApolloHttpException e) {
+      onFailure(e);
+    }
+
+    public void onNetworkError(@Nonnull ApolloNetworkException e) {
```

Comment:
We decided to wrap with ApolloNetworkException that extends ApolloException.
So from our sync API call (`com.apollographql.apollo.ApolloCall#execute`) user should expect only one type of errors `ApolloException`.